mesonbuild/modules/i18n.py

```diff
@@ -12,16 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from os import path
 from .. import coredata, mesonlib, build
 import sys
 
 class I18nModule:
 
+    @staticmethod
+    def _read_linguas(state):
+        linguas = path.join(state.environment.get_source_dir(), state.subdir, 'LINGUAS')
+        try:
+            with open(linguas) as f:
+                return [line.strip() for line in f if not line.strip().startswith('#')]
+        except (FileNotFoundError, PermissionError):
+            return []
+
     def gettext(self, state, args, kwargs):
         if len(args) != 1:
             raise coredata.MesonException('Gettext requires one positional argument (package name).')
         packagename = args[0]
-        languages = mesonlib.stringlistify(kwargs.get('languages', []))
+        languages = mesonlib.stringlistify(kwargs.get('languages', self._read_linguas(state)))
         if len(languages) == 0:
             raise coredata.MesonException('List of languages empty.')
```

Comment:
We're still erroring out if there are no languages. This will stop `meson`.

I think we should just return the POT file target, since it's the only thing that makes sense in this case, and ignore the other targets.